jupyter_client/manager.py

```diff
@@ -490,6 +511,238 @@ def is_alive(self):
             return False
 
 
+class AsyncKernelManager(KernelManager):
+    """Manages kernels in an asynchronous manner """
+
+    @gen.coroutine
+    def _launch_kernel(self, kernel_cmd, **kw):
+        """actually launch the kernel
+
+        override in a subclass to launch kernel subprocesses differently
+        """
+        res = yield gen.maybe_future(launch_kernel(kernel_cmd, **kw))
+        raise gen.Return(res)
```

Comment:
Should we use the async / await syntax instead of decorating methods with `@coroutine` and using `gen.Return`?

Reply:
+:100:

Reply:
Yes - this should be updated - the result of sitting dormant for nearly a year.  I'll try to spend some time with this soon.